minio-admin/src/main/java/io/minio/admin/MinioAdminClient.java

```diff
@@ -0,0 +1,523 @@
+      }
+    }
+
+    private HttpUrl getBaseUrl(String endpoint) {
+      validateNotEmptyString(endpoint, "endpoint");
+      HttpUrl url = HttpUrl.parse(endpoint);
+      if (url == null) {
+        validateHostnameOrIPAddress(endpoint);
+        url = new HttpUrl.Builder().scheme("https").host(endpoint).build();
+      } else {
+        validateUrl(url);
+      }
+
+      return url;
+    }
+
+    public MinioAdminClient.Builder endpoint(String endpoint) {
+      setBaseUrl(getBaseUrl(endpoint));
+      return this;
+    }
+
+    public MinioAdminClient.Builder endpoint(String endpoint, int port, boolean secure) {
+      HttpUrl url = getBaseUrl(endpoint);
+      if (port < 1 || port > 65535) {
+        throw new IllegalArgumentException("port must be in range of 1 to 65535");
+      }
+      url = url.newBuilder().port(port).scheme(secure ? "https" : "http").build();
+
+      setBaseUrl(url);
+      return this;
+    }
+
+    public MinioAdminClient.Builder endpoint(URL url) {
+      validateNotNull(url, "url");
+      return endpoint(HttpUrl.get(url));
+    }
+
+    public MinioAdminClient.Builder endpoint(HttpUrl url) {
+      validateNotNull(url, "url");
+      validateUrl(url);
+      setBaseUrl(url);
+      return this;
+    }
+
+    public MinioAdminClient.Builder region(String region) {
+      validateNullOrNotEmptyString(region, "region");
+      this.region = region;
+      this.regionInUrl = region;
+      return this;
+    }
+
+    public MinioAdminClient.Builder credentials(String accessKey, String secretKey) {
+      this.provider = new StaticProvider(accessKey, secretKey, null);
+      return this;
+    }
+
+    public MinioAdminClient.Builder credentialsProvider(Provider provider) {
+      this.provider = provider;
+      return this;
+    }
+
+    public MinioAdminClient.Builder httpClient(OkHttpClient httpClient) {
+      validateNotNull(httpClient, "http client");
+      this.httpClient = httpClient;
+      return this;
+    }
+
+    public MinioAdminClient build() {
+      validateNotNull(baseUrl, "endpoint");
+      if (httpClient == null) {
+        this.httpClient =
+            new OkHttpClient()
+                .newBuilder()
+                .connectTimeout(DEFAULT_CONNECTION_TIMEOUT, TimeUnit.MINUTES)
+                .writeTimeout(DEFAULT_CONNECTION_TIMEOUT, TimeUnit.MINUTES)
+                .readTimeout(DEFAULT_CONNECTION_TIMEOUT, TimeUnit.MINUTES)
+                .protocols(Arrays.asList(Protocol.HTTP_1_1))
+                .build();
+        String filename = System.getenv("SSL_CERT_FILE");
+        if (filename != null && !filename.isEmpty()) {
+          try {
+            this.httpClient = enableExternalCertificates(this.httpClient, filename);
+          } catch (GeneralSecurityException | IOException e) {
+            throw new RuntimeException(e);
+          }
+        }
+      }
+
+      return new MinioAdminClient(
+          baseUrl, (region != null) ? region : US_EAST_1, provider, httpClient);
```

Comment:
sanity check: region is never supplied in MinIO urls?

Reply:
Region for Admin API is useless @Sam-Kramer - although region can be used it is usually not necessary with MinIO. 

It can be left as is with MinioClient implementation.